[Bug] `qmk flash`  (usbasploader) bootloader retry is not working with avrdude: Version 7.2

### Describe the Bug

avrdude:  Version 7.2 changes the text output from "could not find USB device with" to "cannot find USB device with".   
See: https://github.com/avrdudes/avrdude/commit/e172877724bab4489283a6fc7c83ee95b42e7ada
The bootloader retry mechanism in [platforms/avr/flash.mk](https://github.com/qmk/qmk_firmware/blob/6e6fff866e1179b69f78602e5a791fb1a9e029f4/platforms/avr/flash.mk#L133) grep the old string which breaks the retry.

```
qmk flash -kb coseyfannitutti/discipline -km default  
Ψ Compiling keymap with gmake --jobs=1 coseyfannitutti/discipline:default:flash


QMK Firmware 0.22.11
Making coseyfannitutti/discipline with keymap default and target flash

Generating: .build/obj_coseyfannitutti_discipline_default/src/info_deps.d                           [OK]
Generating: .build/obj_coseyfannitutti_discipline_default/src/default_keyboard.c                    [OK]
avr-gcc (Homebrew AVR GCC 9.4.0) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Generating: .build/obj_coseyfannitutti_discipline_default/src/info_config.h                         [OK]
Generating: .build/obj_coseyfannitutti_discipline_default/src/default_keyboard.h                    [OK]
Compiling: .build/obj_coseyfannitutti_discipline_default/src/default_keyboard.c                     [OK]
Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/bootmagic/bootmagic_lite.c                                                       [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/main.c                                                                           [OK]
Assembling: platforms/avr/xprintf.S                                                                 [OK]
Compiling: platforms/avr/printf.c                                                                   [OK]
Compiling: quantum/mousekey.c                                                                       [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/send_string/send_string.c                                                        [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
Compiling: tmk_core/protocol/host.c                                                                 [OK]
Compiling: tmk_core/protocol/report.c                                                               [OK]
Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
Compiling: platforms/suspend.c                                                                      [OK]
Compiling: platforms/synchronization_util.c                                                         [OK]
Compiling: platforms/timer.c                                                                        [OK]
Compiling: platforms/avr/hardware_id.c                                                              [OK]
Compiling: platforms/avr/platform.c                                                                 [OK]
Compiling: platforms/avr/suspend.c                                                                  [OK]
Compiling: platforms/avr/timer.c                                                                    [OK]
Compiling: platforms/avr/bootloaders/usbasploader.c                                                 [OK]
Compiling: tmk_core/protocol/vusb/protocol.c                                                        [OK]
Compiling: tmk_core/protocol/vusb/vusb.c                                                            [OK]
Compiling: tmk_core/protocol/vusb/usb_util.c                                                        [OK]
Compiling: lib/vusb/usbdrv/usbdrv.c                                                                 [OK]
Assembling: lib/vusb/usbdrv/usbdrvasm.S                                                             [OK]
Compiling: lib/vusb/usbdrv/oddebug.c                                                                [OK]
Linking: .build/coseyfannitutti_discipline_default.elf                                              [OK]
Creating load file for flashing: .build/coseyfannitutti_discipline_default.hex                      [OK]
Copying coseyfannitutti_discipline_default.hex to qmk_firmware folder                               [OK]
Checking file size of coseyfannitutti_discipline_default.hex                                        [OK]
 * The firmware size is fine - 14618/28672 (50%, 14054 bytes free)
Flashing for bootloader: usbasploader
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:404]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:405]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:433]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:434]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:466]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:467]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:488]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:489]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:552]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:553]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:572]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:573]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:591]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:592]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:622]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:623]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:676]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:677]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:760]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:762]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:771]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:773]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:783]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:785]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:796]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:798]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:808]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:809]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:831]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:833]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:855]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:857]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1433]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1434]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1455]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1456]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1469]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1470]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1483]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1484]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:1708]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3199]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3388]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3577]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3769]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3961]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4154]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4340]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4664]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4883]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5083]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5439]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5651]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5905]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:6139]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7104]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7301]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7498]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7755]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7982]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8209]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8438]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8635]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8821]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9008]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9196]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9384]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9725]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9923]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10107]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10390]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10574]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10757]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10943]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11133]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11336]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11778]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11971]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12164]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12480]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12663]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12854]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13045]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13249]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13451]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13452]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13638]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13639]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13827]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13828]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14015]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14016]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14203]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14204]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14387]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14609]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14829]
avrdude warning: has_pdi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15108]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15236]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15325]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15425]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15585]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15643]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15755]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15887]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15987]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16264]
avrdude warning: is_avr32 is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16265]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16289]
avrdude warning: has_tpi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16487]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16644]
avrdude warning: has_updi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16709]
avrdude warning: has_updi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:18001]
avrdude error: cannot find USB device with vid=0x16c0 pid=0x5dc vendor='www.fischl.de' product='USBasp'
avrdude error: unable to open programmer usbasp on port usb

avrdude done.  Thank you.

gmake[1]: *** [platforms/avr/flash.mk:173: flash] Error 1
gmake: *** [Makefile:392: coseyfannitutti/discipline:default:flash] Error 1
Make finished with errors
```

### Keyboard Used

coseyfannitutti/discipline

### Link to product page (if applicable)

_No response_

### Operating System

MacOS 11.7.9

### qmk doctor Output

```text
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.2
Ψ QMK home: /Users/dominik/git/my_1st_keyboard/qmk_firmware
Ψ Detected macOS 11.7.9 (Intel).
Ψ Git branch: master
Ψ Repo version: 0.22.11
Ψ - Latest master: 2023-10-09 01:37:21 -0700 (6e6fff866e) -- Fix typo in `docs/feature_tri_layer.md` (#22232)
Ψ - Latest upstream/master: 2023-10-09 01:37:21 -0700 (6e6fff866e) -- Fix typo in `docs/feature_tri_layer.md` (#22232)
Ψ - Latest upstream/develop: None
Ψ - Common ancestor with upstream/master: 2023-10-09 01:37:21 -0700 (6e6fff866e) -- Fix typo in `docs/feature_tri_layer.md` (#22232)
Ψ - Common ancestor with upstream/develop: None
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 8.3.1
Ψ Found avr-gcc version 9.4.0
⚠ We do not recommend avr-gcc newer than 8. Downgrading to 8.x is recommended.
Ψ Found avrdude version 7.2
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)
Ψ - lib/chibios-contrib: 2023-07-17 11:39:05 +0200 --  (da78eb37)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f)
Ψ QMK is ready to go, but minor problems were found
```


### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

### Other keyboard-related software installed

_No response_

### Additional Context

_No response_

--- a/platforms/avr/flash.mk
+++ b/platforms/avr/flash.mk
@@ -130,10 +130,10 @@ avrdude-split-right: $(BUILD_DIR)/$(TARGET).hex check-size cpfirmware
 	$(call EXEC_AVRDUDE,eeprom-righthand.eep)
 
 define EXEC_USBASP
-	if $(AVRDUDE_PROGRAMMER) -p $(AVRDUDE_MCU) -c usbasp 2>&1 | grep -q "could not find USB device with"; then \
+	if $(AVRDUDE_PROGRAMMER) -p $(AVRDUDE_MCU) -c usbasp 2>&1 | grep -q "\(could not\|cannot\) find USB device with"; then \
 		printf "$(MSG_BOOTLOADER_NOT_FOUND_QUICK_RETRY)" ;\
 		sleep $(BOOTLOADER_RETRY_TIME) ;\
-		until $(AVRDUDE_PROGRAMMER) -p $(AVRDUDE_MCU) -c usbasp 2>&1 | (! grep -q "could not find USB device with"); do\
+		until $(AVRDUDE_PROGRAMMER) -p $(AVRDUDE_MCU) -c usbasp 2>&1 | (! grep -q "\(could not\|cannot\) find USB device with"); do\
 			printf "." ;\
 			sleep $(BOOTLOADER_RETRY_TIME) ;\
 		done ;\

